micropip/package_index.py

```diff
@@ -218,6 +218,10 @@ def _contain_placeholder(url: str, placeholder: str = "package_name") -> bool:
     return placeholder in fields
 
 
+class UnsupportedContentTypeError(Exception):
```

Comment:
Maybe let's put the custom exceptions into a separate file (`error.py`) and import from there.